### PR TITLE
GH-1058 Adds Show/Hide Legend to MegaChart

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes GH-1058

@GordonSmith please review

http://rawgit.com/jbrundage/Visualization/GH-1058-MC-ToggleLegend/demos/dermatology.html?src/composite/MegaChart

Signed-off-by: Jay Brundage <jaman.brundage@lexisnexis.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
Clicking the icon hides the titles as well as the legend...

**Comment by GordonSmith** [comment]
@jbrundage Needs rebase (and see above comment)

**Comment by jbrundage** [comment]
@GordonSmith After some investigation here's what I've found:

Any call to MegaChart.prototype.update causes the same issue (the titles vanish).

The Cell widgets within Border are being recreated on each call to MegaChart.prototype.update. But the X/YAxis labels Text() widgets remain the same **except they're losing their _target elements** so they are never displayed.

Any idea why the Border Cell widgets aren't updating the Text _target when the Cell is recreated?

File: src/composite/MegaChart.js [hunk 1]
```diff
@@ -86,7 +110,7 @@
         } else {
             this._layout.clearContent("bottom");
         }
-        
+        this._layout.render();
```
Comment (GordonSmith) [inline src/composite/MegaChart.js]: Does this need a test for null?  You have one above...
